Formula/q.rb

```diff
@@ -1,30 +1,32 @@
-class Q < Formula
-  include Language::Python::Virtualenv
+# frozen_string_literal: true
 
+# Formula for q
+class Q < Formula
   desc "Run SQL directly on CSV or TSV files"
   homepage "https://harelba.github.io/q/"
-  url "https://github.com/harelba/q/archive/2.0.20.tar.gz"
-  sha256 "46793aef623aac3700856c699cc04810b7a53533f829318729cee900c499a7e1"
-  license "GPL-3.0-or-later"
-  head "https://github.com/harelba/q.git", branch: "master"
+  url "https://github.com/harelba/q/archive/v3.1.6.tar.gz"
+  sha256 "e63ba4ae49647f764c5255ad7065d2c614fdf03a2f7349a795de69529701fab8"
 
-  bottle do
-    sha256 cellar: :any_skip_relocation, arm64_monterey: "c72ca06a7c9dbe3b3eaee1b8db72811edbf7e64fbee5b694bcf2ed7e8d877d50"
-    sha256 cellar: :any_skip_relocation, arm64_big_sur:  "c72ca06a7c9dbe3b3eaee1b8db72811edbf7e64fbee5b694bcf2ed7e8d877d50"
-    sha256 cellar: :any_skip_relocation, monterey:       "4511c183df36704ec7cb497b4a319409875ea2ef6068255ae2a2e0a2d7293e29"
-    sha256 cellar: :any_skip_relocation, big_sur:        "4511c183df36704ec7cb497b4a319409875ea2ef6068255ae2a2e0a2d7293e29"
-    sha256 cellar: :any_skip_relocation, catalina:       "4511c183df36704ec7cb497b4a319409875ea2ef6068255ae2a2e0a2d7293e29"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:   "bed14a331133ff96b85fa37e0729ca695bd273f78ee82e792185d137edf9917a"
-  end
+  license "GPL-3.0-or-later"
+  revision 1
 
+  depends_on "pyoxidizer" => :build
+  depends_on "python@3.8" => :build
   depends_on "ronn" => :build
-  depends_on "python@3.9"
-  depends_on "six"
+  depends_on xcode: ["12.4", :build]
```

Comment:
Why is this needed?

Reply:
pyoxidizer needs this version specifically, because of some bug. Tested on github action without it and it doesn't work. Just a build dependency ofcourse.

Reply:
This will most likely have implications on the macOS versions that can run this software. Not sure what impact it'll have for Linux users.

Reply:
True, haven't thought what would happen for the brewlinux. 

Looking at the current CI - Linux build has been fully successful, and uploaded a bottle, so it must have handled it properly (pyoxidizer doesn't really need this dependency on linux).

Waiting for the other builds to finish to see that they're properly working on 10.15, 11, 12.

Reply:
10.15, 11, 12 and Linux have passed successfully with uploaded bottles

Reply:
Note that this actually means Xcode-12.4-or-newer. It'll be fine for CI, but it also means anyone running Mojave or older won't be able to install this.

Reply:
yeah, thanks. I'm aware of that. I'll issue another fixed version when pyoxidizer fixes their bug.